dashboard/client/src/components/NewEventTable.tsx

```diff
@@ -0,0 +1,371 @@
+import {
+  Box,
+  InputAdornment,
+  LinearProgress,
+  makeStyles,
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TextField,
+  TextFieldProps,
+  Tooltip,
+} from "@material-ui/core";
+import { SearchOutlined } from "@material-ui/icons";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import Pagination from "@material-ui/lab/Pagination";
+import dayjs from "dayjs";
+import React, { useEffect, useState } from "react";
+import { getEvents } from "../service/event";
+import { Event } from "../type/event";
+import { useFilter } from "../util/hook";
+import { StatusChip } from "./StatusChip";
+
+export enum SeverityLevel {
+  INFO = "INFO",
+  DEBUG = "DEBUG",
+  WARNING = "WARNING",
+  ERROR = "ERROR",
+  TRACING = "TRACING",
+} // to maintain and sync with event.proto
+export enum SourceType {
+  COMMON = "COMMON",
+  CORE_WORKER = "CORE_WORKER",
+  GCS = "GCS",
+  RAYLET = "RAYLET",
+  CLUSTER_LIFECYCLE = "CLUSTER_LIFECYCLE",
+  AUTOSCALER = "AUTOSCALER",
+  JOBS = "JOBS",
+  SERVE = "SERVE",
+} // to maintain and sync with event.proto
+
+type EventTableProps = {
+  defaultSeverityLevels?: SeverityLevel[];
+  entityName?: string;
+  entityId?: string; // It could be a specific or "*" to represent all entities
+};
+
+const transformFiltersToParams = (filters: Filters) => {
+  if (!filters) {
+    return null;
+  }
+
+  const params = new URLSearchParams();
+
+  if (filters.entityName && filters.entityId) {
+    params.append(
+      encodeURIComponent(filters.entityName),
+      encodeURIComponent(filters.entityId),
+    );
+  }
+
+  for (const key in filters) {
+    if (key === "sourceType" || key === "severityLevel") {
+      const filterArray = filters[key as keyof Filters] as string[];
+      filterArray.forEach((value) => {
+        params.append(encodeURIComponent(key), encodeURIComponent(value));
+      });
+    } else {
+      // key === 'limit' or other key to add in the future
+      params.append(
+        encodeURIComponent(key),
+        encodeURIComponent(filters[key as keyof Filters] as string),
+      );
+    }
+  }
+
+  return params.toString();
+};
+
+const useStyles = makeStyles((theme) => ({
+  overflowCell: {
+    display: "block",
+    margin: "auto",
+    maxWidth: 360,
+    textOverflow: "ellipsis",
+    overflow: "hidden",
+    whiteSpace: "nowrap",
+  },
+  filterContainer: {
+    display: "flex",
+    alignItems: "center",
+  },
+  search: {
+    margin: theme.spacing(1),
+    display: "inline-block",
+    fontSize: 12,
+    lineHeight: "46px",
+    height: 56,
+  },
+
+  tableContainer: {
+    overflowX: "scroll",
+  },
+
+  helpInfo: {
+    marginLeft: theme.spacing(1),
+  },
+  message: {
+    maxWidth: "200",
```

Comment:
```suggestion
    maxWidth: 200,
```

Reply:
Understood, so we support two ways to express size, 200 or "200px"